internal/config/config.go

```diff
@@ -13,25 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package config
 
 import (
+	"github.com/spf13/viper"
 	"log/slog"
 	"strings"
-
-	"github.com/spf13/viper"
 )
 
-func InitViper() error {
-	viper.AutomaticEnv()
-	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
-	viper.AddConfigPath(".")
-	viper.SetConfigFile("./config.yaml")
+type ResourceConfig struct {
+	Type         string
+	NestedConfig map[string]interface{} `mapstructure:",remain"`
```

Comment:
What about something like `LocationConfig` or `EndpointConfig`? I find `NestedConfig` to be somewhat generic

Reply:
😅 In the documentation this field was always named `Other`, because this field contains all the other fields.
I'm open to rename it, but Location or Endpoint doesn't match all our resource types. 🤔

Another idea could be `ResourceSpecificConfig` but in my opinion this is way to long 😕

We could also split the struct in two different ones (I guess we will do that in the future anyway) and use then `ResourceConfig` for the struct field:

```go
type SourceConfig struct {
	Type                   string
	ResourceConfig map[string]any `mapstructure:",remain"`
}
type DestinationConfig struct {
	Type                   string
	ResourceConfig map[string]any `mapstructure:",remain"`
}
```

WDYT?

Reply:
I like the splitting into distinct configs

Reply:
done